r/R/dplyr-collect.R

```diff
@@ -27,6 +27,10 @@ collect.arrow_dplyr_query <- function(x, as_data_frame = TRUE, ...) {
       augment_io_error_msg(e, call, schema = x$.data$schema)
     }
   )
+  # Ungroup the Table before converting this to a data frame
```

Comment:
I wonder if it would be simpler to move the `ungroup()` logic to `arrow_dplyr_query()` (specifically, around L48 in dplyr.R). That way, we know that there are no lingering groups in metadata anywhere once we're doing a dplyr query, the single source of truth is the `arrow_dplyr_query` object.

Reply:
Thanks for the suggestion.
I assumed that `arrow_dplyr_query$.data` must have the original object stored as is.

I will update it as you suggested, as it seems to pass the test!